src/eos-metrics-instrumentation.c

```diff
@@ -284,6 +300,72 @@ record_live_boot (gpointer unused)
     return G_SOURCE_REMOVE;
 }
 
+static gchar *
+get_image_version_for_path (const gchar *path)
+{
+    ssize_t xattr_size;
+    gchar *image_version = NULL;
+
+    xattr_size = getxattr (path, EOS_IMAGE_VERSION_XATTR, NULL, 0);
+
+    if (xattr_size < 0)
```

Comment:
I think you might as well put the overflow check in. If we’re going to be paranoid enough to check for valid UTF-8, we might as well also check for overflow.